app/server/models/user.py

```diff
@@ -408,6 +410,30 @@ def decode_single_use_JWS(cls, token, required_type):
 
             return {'success': False, 'message': e}
 
+    def save_password_reset_token(self, password_reset_token):
+        # make a "clone" of the existing token list
+        self.clear_expired_tokens()
+        current_reset_tokens = self.password_reset_tokens[:]
+        current_reset_tokens.append(password_reset_token)
+        # set db value
+        self.password_reset_tokens = current_reset_tokens
```

Comment:
we don't have to do `db.session.commit()` for these kinds of things?

Reply:
oh i see one at the end of the reset password api call, but not one at the end of request password reset email one. have you tested this works as expected?

Reply:
That's what i thought too. But it get stored in the db without it

Reply:
That's because of this function that is executed after a successful request

https://github.com/teamsempo/SempoBlockchain/blob/381f82424b1eec8fa1d6446f84a1099e5b039575/app/server/__init__.py#L158-L163